changesets2CSV/changesets2CSV

```diff
@@ -121,6 +173,308 @@ def count_new_modified_deleted(changeset):
+        return max(parsed_response['wait_time'])
+    return wait_time
+
+def overpass_query(query):
+    """Query the overpass servers. This may block for extended periods of time, depending upon the query"""
+
+    session = requests.session()
+    session.headers.update({'User-Agent': OVERPASS_USER_AGENT})
+    cached_session = CacheControl(session)
+    response = cached_session.post("http://overpass-api.de/api/interpreter", data={'data': query})
+    wait_time = overpass_status()
+    loop = 0
+    while (wait_time > 0):
+        print('Waiting for wait_time: ',wait_time)
+        time.sleep(wait_time)
+        wait_time = overpass_status()
+        loop += 1
+    while (response.status_code == requests.codes.too_many_requests) or (response.status_code == requests.codes.gateway_timeout):
+        print('Waiting to resubmit . . .', end="\r")
+        time.sleep(10)
+        print('Resubmitted')
+        response = cached_session.post("http://overpass-api.de/api/interpreter", data={'data': query})
+    if (response.status_code != requests.codes.ok):
+        print("Bad request")
+        print(response.text)
+        print(response.status_code)
+        raise ValueError("Bad Request: {}".format(query))
+
+    xml = response.text
+
+    if (response.status_code != requests.codes.ok):
+        raise ValueError("We got a bad response code of {} for {} which resulted in:\r\n{}".format(response.status_code, query, xml))
+    content_type = response.headers.get('content-type')
+    if content_type == 'application/osm3s+xml':
+        return ET.ElementTree(ElementTree.fromstring(xml))
+    elif content_type == 'application/json':
+        return response.json()
+    else:
+        raise ValueError("Unexpected content type ({}) from the query: {}".format(content_type, query))
+
+def count_tag_change(changesets,info_json,osm_obj_type='*'):
+    tags_to_check = info_json['tags']
+
+    #Testing variables
+    print_version_lists = False
+    object_limit_for_query=0
+    print_query = False
+    dont_run_query = False
+    record_last_result = True
+    use_hardcoded_query_result = False
+    dont_process_query = False
+    note_every_change = False
+
+
+
+    #TODO:
+    #We could change this to: {<changeset_id>:{<tag_to_check>:[<objects>],'add'':0,'modify':0,'delete':0}}
+    #Then we could just get len()s of objs_modified, objs_created, objs_deleted
+
+    objects_by_changeset = {}
+    #Formatted as {<changeset_id>:{<tag_constant>:[<objects>]}}
+
+    #Get all objects touched in each changeset
+    #We go by changeset, then by tag
+    for changeset in changesets:
+        #New changeset in list
+        objects_by_changeset[changeset] = {}
+        #Request XML for each changeset
+        api_url = "https://www.openstreetmap.org/api/0.6/changeset/{changeset}/download".format(changeset=changeset)
+        dev_api_url = "https://master.apis.dev.openstreetmap.org/api/0.6/changeset/{changeset}/download".format(changeset=changeset)
```

Comment:
> what are we using this for?
> 
> If we want to keep all the debug/testing outputs it may be best to include a "debug" flag. If we do that we could do something like this for this scenario:
> 
> ```
> if debug:
>      api_url = xxxxxx
> else:
>      api_url = xxxxx
> ```

I actually never ended up using the dev api, since I just had a bunch of changeset info from the company that I could use instead. Will remove